packages/browser/src/integrations/globalhandlers.ts

```diff
@@ -52,206 +55,202 @@ export class GlobalHandlers implements Integration {
   /**
    * @inheritDoc
    */
-  public setupOnce(): void {
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
     Error.stackTraceLimit = 50;
 
-    if (this._options.onerror) {
-      logger.log('Global Handler attached: onerror');
-      this._installGlobalOnErrorHandler();
-    }
-
-    if (this._options.onunhandledrejection) {
-      logger.log('Global Handler attached: onunhandledrejection');
-      this._installGlobalOnUnhandledRejectionHandler();
-    }
-  }
-
-  /** JSDoc */
-  private _installGlobalOnErrorHandler(): void {
-    if (this._onErrorHandlerInstalled) {
-      return;
+    const hub = getCurrentHub();
+    const client = hub.getClient();
+    const attachStacktrace = client && client.getOptions().attachStacktrace;
+    const options = this._options;
+
+    // We can disable guard-for-in as we construct the options object above + do checks against
+    // `this._installFunc` for the property.
+    // eslint-disable-next-line guard-for-in
+    for (const key in options) {
+      const installFunc = this._installFunc[key as GlobalHandlersIntegrationsOptionKeys];
+      if (installFunc && options[key as GlobalHandlersIntegrationsOptionKeys]) {
+        globalHandlerLog(key);
+        installFunc(hub, attachStacktrace);
+        this._installFunc[key as GlobalHandlersIntegrationsOptionKeys] = undefined;
```

Comment:
tbh idk why have this guard because `addInstrumentationHandler` should be sufficient, but keeping it here for now 🤷